validate.go

```diff
@@ -1,16 +1,232 @@
+	if err != nil {
+		return nil, fmt.Errorf("Cannot get namespace data")
+	}
+	namespace := &corev1.Namespace{}
+	if err := easyjson.Unmarshal(responseBytes, namespace); err != nil {
+		return nil, fmt.Errorf("Cannot parse namespace data")
+	}
+	return namespace, nil
+}
+
+func validateResourceLabels(namespaceLabels map[string]string, request kubewarden_protocol.ValidationRequest, settings Settings) ([]byte, error) {
+	newLabels := make(map[string]string)
+	for _, label := range settings.PropagatedLabels {
+		if value, namespace_has_label := namespaceLabels[label]; namespace_has_label {
+			newLabels[label] = value
+		}
+	}
+	return updateResourceLabels(request, newLabels)
+
+}
+func getResourceMetadata(object kubewarden_protocol.ValidationRequest) (*apimachineryv1.ObjectMeta, error) {
+	switch object.Request.Kind.Kind {
+	case DEPLOYMENT_KIND:
+		deployment := appsv1.Deployment{}
+		if err := easyjson.Unmarshal(object.Request.Object, &deployment); err != nil {
+			return nil, err
+		}
+		return deployment.Metadata, nil
+	case REPLICASET_KIND:
+		replicaset := appsv1.ReplicaSet{}
+		if err := easyjson.Unmarshal(object.Request.Object, &replicaset); err != nil {
+			return nil, err
+		}
+		return replicaset.Metadata, nil
+	case STATEFULSET_KIND:
+		statefulset := appsv1.StatefulSet{}
+		if err := easyjson.Unmarshal(object.Request.Object, &statefulset); err != nil {
+			return nil, err
+		}
+		return statefulset.Metadata, nil
+	case DAEMONSET_KIND:
+		daemonset := appsv1.DaemonSet{}
+		if err := easyjson.Unmarshal(object.Request.Object, &daemonset); err != nil {
+			return nil, err
+		}
+		return daemonset.Metadata, nil
+	case REPLICATIONCONTROLLER_KIND:
+		replicationController := corev1.ReplicationController{}
+		if err := easyjson.Unmarshal(object.Request.Object, &replicationController); err != nil {
+			return nil, err
+		}
+		return replicationController.Metadata, nil
+	case CRONJOB_KIND:
+		cronjob := batchv1.CronJob{}
+		if err := easyjson.Unmarshal(object.Request.Object, &cronjob); err != nil {
+			return nil, err
+		}
+		return cronjob.Metadata, nil
+	case JOB_KIND:
+		job := batchv1.Job{}
+		if err := easyjson.Unmarshal(object.Request.Object, &job); err != nil {
+			return nil, err
+		}
+		return job.Metadata, nil
+	case POD_KIND:
+		pod := corev1.Pod{}
+		if err := easyjson.Unmarshal(object.Request.Object, &pod); err != nil {
+			return nil, err
+		}
+		return pod.Metadata, nil
+	default:
+		return nil, fmt.Errorf("object should be one of these kinds: Deployment, ReplicaSet, StatefulSet, DaemonSet, ReplicationController, Job, CronJob, Pod. Found %s", object.Request.Kind.Kind)
+	}
+}
+
+func updateLabels(labels *map[string]string, newLabels map[string]string) bool {
+	has_mutation := false
+	for label, newValue := range newLabels {
+		if oldValue, has_label := (*labels)[label]; !has_label || oldValue != newValue {
+			(*labels)[label] = newValue
+			has_mutation = has_mutation || true
```

Comment:
simpler, more direct
```suggestion
			has_mutation = true
```